### ISSUE TITLE
Clickhouse Bad Address

### ISSUE THREAD

**Issue by ColeTownsend** [railwayapp-templates/plausible#7]
## Summary
On a fresh deploy of Plausible, the main docker image side throws an error `nc: bad address 'clickhouse.railway.internal` You will be able to log in to the dashboard and it will receive data about visitors, but the main page will get stuck.

<img width="1140" alt="image" src="https://github.com/railwayapp-templates/plausible/assets/987654/98a4fc1f-5d80-4e5c-875b-8ebb94d5ec59">

## Recreation
Deploy a brand new Plausible

## Logs
Clickhouse
```
/entrypoint.sh: create new user 'default' instead 'default'

Processing configuration file '/etc/clickhouse-server/config.xml'.

Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.

Merging configuration file '/etc/clickhouse-server/config.d/main-config.xml'.

Logging trace to /var/log/clickhouse-server/clickhouse-server.log

Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log

/entrypoint.sh: create database 'plausible_events_db'

Processing configuration file '/etc/clickhouse-server/config.xml'.

Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.

Merging configuration file '/etc/clickhouse-server/config.d/main-config.xml'.

Logging trace to /var/log/clickhouse-server/clickhouse-server.log

Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log

/entrypoint.sh: create new user 'default' instead 'default'

ClickHouse Database directory appears to contain a database; Skipping initialization

Processing configuration file '/etc/clickhouse-server/config.xml'.

Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.

Merging configuration file '/etc/clickhouse-server/config.d/main-config.xml'.

Logging trace to /var/log/clickhouse-server/clickhouse-server.log

Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log
```

**Comment by MykalMachon** [comment]
Hey @ColeTownsend, thanks for creating this issue. 
The NC bad address thing is actually all good! It's intentional behavior and is just plausible waiting for clickhouse db to come up and online.

That said, the issues your reporting and that I'm seeing on my side seem to be [a known issue caused by a recent release of Plausible](https://github.com/ClickHouse/ClickHouse/issues/60685), they have a workaround and a fix will be coming in soon.

I'll implement the workaround now, it may just require you to make some changes to your clickhouse config in the repo forked as part of the railway template deploy. Once I have a fix in place I'll send you instructions.

**Comment by MykalMachon** [comment]
Okay, @ColeTownsend  I've confirmed that the fix in #8 got it working again on a new deploy of Plausible.
All you need to do to get your instance working is either 

### Option 1: copy over the clickhouse config file into your forked repo 
When you deployed this template, railway created a fork of this repo for you on your GitHub account called something like "Plausible". You just need to find that repo and merge the changes in this repo (the upstream) back down into your fork. 

If you're not sure how to do that, you can also just replace the `https://github.com/railwayapp-templates/plausible/blob/main/packages/clickhouse/ch-user-config.xml` in your plausible repo with the file in this plausible repo via a new commit.

Let me know if you have any issues with that.

### Option 2: Delete your current project and redeploy via the template
This will delete all your data and is definitely the less optimal option but the quickest. I would only recomend this if you don't have any data in plausible yet. 

Enjoy the rest of your week!
